[ADVISOR-2723] Add RBAC

johnsonm325 opened the pull request:
This PR adds RBAC security to the frontend. At this point, we only have `tasks:*:*`, `inventory:*:*` and `inventory:*:read`. I block access to every route of the app if a user doesn't have `tasks:*:*` and I only block the SystemsTable when a user doesn't have `inventory:*:*` OR `inventory:*:read`.

If you don't have experience setting up RBAC permissions, I can help with showing you how to do it in order to properly test this. You can surely just set `hasAccess` manually in the code, but that doesn't quite have the same effect as setting RBAC perms. Let me know if you need guidance.

bastilian commented on src/SmartComponents/RunTaskModal/RunTaskModal.js:
```diff
@@ -24,6 +26,10 @@ const RunTaskModal = ({
   title,
 }) => {
   const [selectedIds, setSelectedIds] = useState(selectedSystems);
+  const { hasAccess, isLoading } = usePermissions('inventory', [
```
The inventory itself should/is checking these permissions, or does it not?

johnsonm325 commented:
@bastilian You can take another look at this if you want. I removed the inventory check in the `RunTaskModal`, but I added a separate inventory check in the `CompletedTaskDetails` component to check inventory permissions to display job details for a completed task.

bastilian commented on src/PresentationalComponents/WithPermission/__tests__/WithPermission.tests.js:
```diff
@@ -0,0 +1,78 @@
+jest.mock('@redhat-cloud-services/frontend-components-utilities/RBACHook');
+
+describe('WithPermission', () => {
+  let props;
+
+  beforeEach(() => {
+    props = {
+      requiredPermissions: ['tasks:*:*'],
+    };
+  });
```
This can just be 

```js
const props = {
  requiredPermissions: ['tasks:*:*'],
};
```
, can it not?
